### ISSUE TITLE
INPC001 shouldn't trigger for classes not implementing INPC

### ISSUE THREAD

**Issue by mterwoord** [DotNetAnalyzers/PropertyChangedAnalyzers#125]
Following code triggers INPC001 for property `Test`:

```csharp
    public class MyObject: IEnumerable<string>
    {
        public string Test
        {
            get;
            set;
        }
        
        IEnumerator IEnumerable.GetEnumerator()
        {
            return GetEnumerator();
        }

        public IEnumerator<string> GetEnumerator()
        {
            yield break;
        }
    }
```

**Comment by JohanLarsson** [comment]
Not sure how you mean here. The design of `INPC001` is to nag when a class has mutable properties.
The reason for this design is so that we can know that all properties can be bound to or listened if using reactive extensions.

If you find the default behavior annoying you can configure severity:
-`Hidden` means no output in the error list but the code fix will be available when the cursor is on the identifier of the class. 
- `None` the rule is not checked at all.
- `Info`, `Warning` and `Error` are self descriptive I think.

![image](https://user-images.githubusercontent.com/1640096/63260983-aae23800-c282-11e9-9d50-774b3f50eeb0.png)

Doing ^ adds it to the `.ruleset` file that you add to source control meaning the same settings will be used for all team members.

**Comment by mterwoord** [comment]
Hmm, must have missed the documentation. Sorry for that. So basically disabling `INPC001` stops the compiler from complaining about classes not implementing INPC, but does warn me of the rest?

**Comment by JohanLarsson** [comment]
We can probably improve the docs here.
Also is configuring severity a workable solution?

**Comment by mterwoord** [comment]
It was a misunderstanding of the docs. Changing the severity is definitely a workable sollution. I'm closing the issue now.
